pkg/security/security_profile/activity_tree/activity_tree.go

```diff
@@ -542,24 +602,44 @@ func (at *ActivityTree) findBranch(children *[]*ProcessNode, siblings *[]*Proces
 		// directly exec-ed
 		break
 	}
-	return nil, false
+	return nil, 0, false
 }
 
 // rebaseTree rebases the node identified by "nodeIndexToRebase" in the input "tree" onto a newly created branch made of
 // "branchToInsert" and appended to "treeToRebaseOnto". New nodes will be tagged with the input "generationType".
 // This function returns the top level node, owner of the newly inserted branch that lead to the rebased node
-func (at *ActivityTree) rebaseTree(tree *[]*ProcessNode, treeIndexToRebase int, treeToRebaseOnto *[]*ProcessNode, branchToInsert []*model.ProcessCacheEntry, generationType NodeGenerationType, resolvers *resolvers.Resolvers) *ProcessNode {
+func (at *ActivityTree) rebaseTree2(parent ProcessNodeParent, childIndexToRebase int, newParent ProcessNodeParent, branchToInsert []*model.ProcessCacheEntry, generationType NodeGenerationType, resolvers *resolvers.Resolvers) *ProcessNode {
+	if len(branchToInsert) > 1 {
+		// We know that the entries in "branch" are all "isExecChild = true" nodes, except the top level entry that might be
+		// a "isExecChild = false" node. Similarly, all the nodes below parent.GetChildren()[childIndexToRebase] must be non
+		// matching "isExecChild = true" nodes, except parent.GetChildren()[childIndexToRebase] that might be a "isExecChild
+		// = false" node. To be safe, check if the 2 top level nodes match if one of them is an "isExecChild = true" node.
+		childToRebase := (*parent.GetChildren())[childIndexToRebase]
+		if topLevelNode := branchToInsert[len(branchToInsert)-1]; !topLevelNode.IsExecChild || !childToRebase.Process.IsExecChild {
+			if childToRebase.Matches(&topLevelNode.Process, at.differentiateArgs) {
+				// ChildNodeToRebase and topLevelNode match and need to be merged, rebase the one in the profile, and insert
+				// the remaining nodes of the branch on top of it
+				newRebasedChild := at.rebaseTree2(parent, childIndexToRebase, newParent, nil, generationType, resolvers)
+				output, _, _ := at.insertBranch(newRebasedChild, branchToInsert[:len(branchToInsert)-1], generationType, false, resolvers)
+
+				if output == nil {
+					return newRebasedChild
+				}
+				return output
+			}
+		}
+	}
 
 	// create the new branch
 	var rebaseRoot, childrenCursor *ProcessNode
-	for i := len(branchToInsert) - 1; i >= 0; i-- {
+	for i := len(branchToInsert) - 1; i >= 1; i-- {
```

Comment:
At the first glance, it appears that you do not want to insert the leaf and it should break things. But what's the real reason?